### ISSUE TITLE
User can't change funding choices

### ISSUE THREAD

**Issue by alanna** [cobudget/cobudget#594]
A user named Caroline in [our group](https://cobudget.com/oc/ocf-mid22) funded some buckets, but now she wants to change her mind. How can she undo? If there's no feature for this yet, can her bucket funding be reset so she can start again?

**Comment by aerugo** [comment]
Hi Alanna,

As of now, this is a conscious choice.

My view is that the predictability for the fundee is more important than
the freedom for the funder.

In a way, I would also like the software to be opinionated in favor of
people who create good project proposals early in a funding round, and not
have them worry that if something new and flashy comes around later, they
will see their funding evaporate. Proposing projects in a public setting
and having them be voted on can be intimidating enough, especially in
situations where those projects may be creative or artistic. I think that
it can be quite anxiety-inducing if your progress bar suddenly starts
ticking backwards.

This said, I think that it is different between groups. I would prefer for
non-retractable funding to be the default, but I also see that perhaps we
could allow for a toggle that allows it. I think that in the cases where
Cobudget is used for more classic participatory budgeting - i.e. where it
is participatory prioritization between different communally owned
initiatives, rather than participatory grant distribution towards
individual passion projects, it makes a lot of sense to have retractable
funding be the default behaviour. So far, I think the most common use case
for cobudget has been the latter.

I would be very interested to hear further thoughts on this! Thank you for
bringing it up.

**Comment by aerugo** [comment]
Thinking about this a bit more, I think that a good compromise might be to have an option for moderators to revert transactions to buckets that are not yet fully funded. This would allow groups that need this to either make everyone a moderator, or to ask users to contact a moderator to do this for them if necessary.

**Comment by alanna** [comment]
OK can you manually undo Caroline's funding? It was not clear at all that these choices were going to be permanent and now she's stuck.

I have to say I disagree with your framing here. For example, the reason we always made votes changeable in Loomio is because the process of understanding others' views and evolving your own decisions in response is the heart of participatory process. Otherwise, we should just send out a survey and share the results at the end. I view Cobudget as a "many to many" process, not a "many to one" process, and changing decisions over time is an important part of it. If you can't update your funding choices, it's not really an evolving conversation, but just a poll.

Also, people just make mistakes sometimes and need to fix them. Overall I think the default should be to give users power over their money and trust them to make good choices.

All that said, a toggle seems like a good idea, as limited choice is also an important facilitation tool sometimes.

Until you build such a toggle, can you please make a big warning about funding decisions being final? I also assumed you could change them and was surprised you couldn't, so users need to be informed.

**Comment by francesca-sp** [comment]
@alanna I disagree actually, and am with Hugi on this one. As a note, Cobudget (legacy) has always worked this way until now (this is how I "inherited it"), meaning that funds cannot be retracted. I think that it's quite important to give moderators the ability to retract funds on buckets that are not fully funded, as a rare occasion when people make a mistake or did not realize their funding was permanent (which we know does happen, based on user requests). However I see it was very risky to make it as easy for people to move funds between buckets as it is to change your loomio vote. In an consent process it makes A LOT of sense, because you want people to evolve their thinking - but that's not what Cobudgeting is. From having observed the group dynamics around this, I think that people being swayed quickly by "trends" and suddenly draining funds out of buckets that had a lot of funds is problematic. And since users can "accept funding" at any time to close their bucket, this could then lead to people trying to lock in funds by quickly accepting, if they start to see that happening.

I'm all for adding more options around this in the future, but I don't think that this is an essential short term priority.

**Comment by alanna** [comment]
How about something like you can take back your funding for 24 hours, to deal with the case where someone makes a genuine mistake?

**Comment by alanna** [comment]
In any case, can Caroline's funding choices be undone on the back end or something? She had no idea they weren't changeable and it's blocking her genuine participation in our current round. @aerugo @Powersource

**Comment by aerugo** [comment]
Let me have a look, can you give me the direct URL to the bucket she
funded? Send it to me in an email or post the URL here.

**Comment by alanna** [comment]
I actually don't have your email! I will DM you on the OC Slack @aerugo
